test/e2e/testdata/retry-test.yaml

```diff
@@ -0,0 +1,33 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  name: retry-test
+  labels:
+    argo-e2e: true
+spec:
+  entrypoint: steps-outer
+  templates:
+    - name: steps-outer
+      steps:
+        - - name: steps-outer-step1
+            template: whalesay
+        - - name: steps-outer-step2
+            template: steps-inner
+
+    - name: steps-inner
+      retryStrategy:
+        restartOnWorkflowRetry: true
```

Comment:
My initial reaction is that this is not the correct place for this.

`retryStrategy` deals with retrying this node during a single workflow execution. Your proposed flag deals with retrying the node across different executions. Under this implementation `retryStrategy` is overloaded.